Web Inspector: Fonts Panel: ITAL variation axis slider has value of NaN

rcaliman-apple opened the pull request:
#### c58b326609c90f5e3f07a21a6e2dce4dfd112904
<pre>
Web Inspector: Fonts Panel: ITAL variation axis slider has value of NaN
<a href="https://bugs.webkit.org/show_bug.cgi?id=250906">https://bugs.webkit.org/show_bug.cgi?id=250906</a>

Reviewed by Patrick Angle.

The value of the `font-style` CSS property maps to either
`ITAL` or `SLNT` font variation axes when available.

`font-style: normal` should equate to `ITAL: 0`.
This case wasn&apos;t handled for the `ITAL` axis.

* LayoutTests/inspector/model/font-styles-conversion-expected.txt: Added.
* LayoutTests/inspector/model/font-styles-conversion.html: Added.
* Source/WebInspectorUI/UserInterface/Models/FontStyles.js:
(WI.FontStyles.axisValueToFontPropertyValue):
(WI.FontStyles.fontPropertyValueToAxisValue):
* Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js:
(WI.FontVariationDetailsSectionRow):
(WI.FontVariationDetailsSectionRow.prototype._getAxisResolution):

Canonical link: <a href="https://commits.webkit.org/259351@main">https://commits.webkit.org/259351@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f4934a4eb84a3d8eac701f428245acdeadc8526

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104701 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13778 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37604 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113976 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174181 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108617 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14905 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4709 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97031 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112904 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110462 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/11504 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/94536 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39056 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/108171 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/93364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/26150 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80723 "Passed tests") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/94637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7133 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/27506 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/92594 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/83/builds/4885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7239 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4081 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30154 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103525 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13286 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/47065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101280 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6447 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9026 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/25220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

patrickangle commented on Source/WebInspectorUI/UserInterface/Models/FontStyles.js:
```diff
@@ -86,6 +86,11 @@ WI.FontStyles = class FontStyles
+            if (value === "normal")
+                return 0;
+
             return parseFloat(value);
```
Should this case just use the same logic as `slnt`? Otherwise it seems like we would still fail to parse the oblique keyword/degree values?

patrickangle replied:
Also do we need to similarly support `ital` in the other direction in `fontPropertyValueToAxisValue`?

rcaliman-apple replied:
Yes, you are right. The cases for `ital` and `slnt` have significant overlap. I merged the two and adapted accordingly. 

Also added mapping `ital` value to `font-style: italic | normal` in `fontPropertyValueToAxisValue` under the assumptions [per spec](https://w3c.github.io/csswg-drafts/css-fonts/#valdef-font-style-oblique-angle--90deg-90deg) that an `ital` axis is supposed to act as an on/off switch and its value is expected to be in the range 0–1 (0 = off, 1 = on). 

Checked with @litherum and the guidance is to not do anything special for non-conforming fonts with ranges beyond 0–1. Chose to map anything `ital` value >=1  to `font-style: italic`

Added tests for these mapping functions as they become more complex.

patrickangle commented on Source/WebInspectorUI/UserInterface/Models/FontStyles.js:
```diff
@@ -77,25 +77,38 @@ WI.FontStyles = class FontStyles
 
     static fontPropertyValueToAxisValue(tag, value)
     {
+
```
Oops?

patrickangle commented on LayoutTests/inspector/model/font-styles-conversion.html:
```diff
@@ -0,0 +1,159 @@
+        name: "WI.FontStyles.Conversion.axisValueToFontPropertyValue",
+        description: "Check that font variation axis values map to appropriate font CSS properties values.",
+        test() {
+            let tests = [
```
This should probably be `const`?

patrickangle commented on LayoutTests/inspector/model/font-styles-conversion.html:
```diff
@@ -0,0 +1,159 @@
+        name: "WI.FontStyles.Conversion.fontPropertyValueToAxisValue",
+        description: "Check that font CSS property values map to appropriate font variation axis values.",
+        test() {
+            let tests = [
```
Ditto :15

rcaliman-apple committed:
Web Inspector: Fonts Panel: ITAL variation axis slider has value of NaN
https://bugs.webkit.org/show_bug.cgi?id=250906

Reviewed by Patrick Angle.

The value of the `font-style` CSS property maps to either
`ITAL` or `SLNT` font variation axes when available.

`font-style: normal` should equate to `ITAL: 0`.
This case wasn't handled for the `ITAL` axis.

* LayoutTests/inspector/model/font-styles-conversion-expected.txt: Added.
* LayoutTests/inspector/model/font-styles-conversion.html: Added.
* Source/WebInspectorUI/UserInterface/Models/FontStyles.js:
(WI.FontStyles.axisValueToFontPropertyValue):
(WI.FontStyles.fontPropertyValueToAxisValue):
* Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js:
(WI.FontVariationDetailsSectionRow):
(WI.FontVariationDetailsSectionRow.prototype._getAxisResolution):

Canonical link: https://commits.webkit.org/259351@main
